pandas/core/groupby/generic.py

```diff
@@ -1119,7 +1119,7 @@ def _aggregate_frame(self, func, *args, **kwargs) -> DataFrame:
         axis = self.axis
         obj = self._obj_with_exclusions
 
-        result: OrderedDict = OrderedDict()
+        result: dict = {}
```

Comment:
We don't use the builtins for annotations, rather want to import `Dict` from typing. Would be extra useful if you can figure out the subtypes that need to be added

Reply:
Sure - I ran the tests cases that touch this logic and think I've now got the right subtypes